src/mlpack/methods/ann/layer/repeat_impl.hpp

```diff
@@ -0,0 +1,204 @@
+RepeatType<MatType>::RepeatType(RepeatType&& other)  noexcept :
+    Layer<MatType>(std::move(other)),
+    multiples(other.multiples),
+    interleave(other.interleave),
+    outIdxs(other.outIdxs),
+    sizeMult(other.sizeMult),
+    backIdxs(other.backIdxs)
+{
+  // Nothing else to do.
+}
+
+template<typename MatType>
+RepeatType<MatType>& RepeatType<MatType>::operator=(const RepeatType& other)
+{
+  if (this != &other)
+  {
+    Layer<MatType>::operator=(other);
+    multiples = other.multiples;
+    interleave = other.interleave;
+    outIdxs = other.outIdxs;
+    sizeMult = other.sizeMult;
+    backIdxs = other.backIdxs;
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+RepeatType<MatType>& RepeatType<MatType>::operator=(RepeatType&& other) noexcept
+{
+  if (this != &other)
+  {
+    Layer<MatType>::operator=(std::move(other));
+    multiples = std::move(other.multiples);
+    interleave = std::move(other.interleave);
+    outIdxs = std::move(other.outIdxs);
+    sizeMult = other.sizeMult;
+    backIdxs = other.backIdxs;
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+void RepeatType<MatType>::ComputeOutputDimensions()
+{
+  if (multiples.size() > this->inputDimensions.size())
+  {
+    std::ostringstream oss;
+    oss << "Repeat::ComputeOutputDimensions(): multiples vector must "
+        << "have the same or fewer dimensions than InputDimensions";
+    throw std::invalid_argument(oss.str());
+  }
+
+  size_t inputSize = this->inputDimensions[0];
+  for (size_t i = 1; i < this->inputDimensions.size(); i++)
+  {
+    inputSize *= this->inputDimensions[i];
+  }
+  arma::umat idxs = arma::regspace<arma::uvec>(0, inputSize - 1);
+
+  // Here, we are going to pre-compute the source index for each output
+  // for a single tensor.  Since the tensors are flattened into 1-d
+  // vectors, we can fill the output row-wise based on these
+  // indices.
+  this->outputDimensions = this->inputDimensions;
+  sizeMult = 1;
+  size_t outSize = 1;
+
+  // iteratively reshape the index matrix such that the dimension
+  // to be repeated (and all prior) are flattened to a column, and
+  // then repelem rowwise or repmat columwise.
+  for (size_t i = 0; i < multiples.size(); i++)
+  {
+    if (multiples[i] != 1)
+    {
+      if (interleave)
+      {
+        // For the first dimension, we need to do the repelem columnwise.
+        if (i == 0)
+        {
+          idxs.reshape(outSize * this->inputDimensions[i],
+                       idxs.n_elem / (outSize * this->inputDimensions[i]));
+          idxs = arma::repelem(idxs, multiples[i], 1);
+        }
+        else
+        {
+          idxs.reshape(outSize,
+                       idxs.n_elem / outSize);
+          idxs = arma::repelem(idxs, 1, multiples[i]);
+        }
+      }
+      else
+      {
+        idxs.reshape(outSize * this->inputDimensions[i],
+                     idxs.n_elem / (outSize * this->inputDimensions[i]));
+        idxs = arma::repmat(idxs, multiples[i], 1);
```

Comment:
Would it be possible to remove the `arma::` since we are working on bandicoot integration, it would be nice to have the new merged code already compatible with what are we doing right now.